joomla/joomla-cms
# Button cleanup

Pull Request for Issue # .
### Summary of Changes
This cleans up mainly modal-related buttons. Most notable change is restoration of `<button>` element which was removed in PRs #11787, #14483, #16156. The original issue #11765 was caused by missing `type` attribute in user field layout. Although switching to `<a>` tag fixed the issue, it wasn't the right way to do it.

For future reference:
Use `<button>` element in favor of `<a>` for button functionality whenever possible (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_button_role).
Always add `type` attribute (most browsers default to `type="submit"` without it).
`button` is not a valid value for `type` attribute on `<a>` (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-type).
Don't use `aria-hidden="true"` on buttons containing readable text. Use it on non-readable elements like icons.

### Testing Instructions
1. Code review.
2. Check that buttons work correctly.
3. Confirm that you cannot reproduce issue #11765.
### Documentation Changes Required
No.
